### ISSUE TITLE
Creating cluster Fails 400/EULA_NOT_ACCEPTED

### ISSUE THREAD

**Issue by frenchtoasters** [vmware/container-service-extension#159]
I have setup my cse server following the guide by @tsugliani here https://cloud.tsugliani.fr/2018/07/vcloud-director-and-cse-part-3/ I am able to run all the display commands without issue however when creating a cluster using the default template I am getting the following EULA error:

```
root@photon-machine [ ~ ]# vcd cse cluster create -N 3 -n orgNetwork clusterName --enable-nfs
InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised.
create_cluster: Creating cluster TylerWasHere(375cd040-d3b7-4645-aac5-c29c49a6f200)
create_cluster: Creating cluster vApp TylerWasHere(375cd040-d3b7-4645-aac5-c29c49a6f200)
task: 5a0239f2-bc0d-4d03-afd2-ab09989be95d, result: error, message: Status code: 400/EULA_NOT_ACCEPTED, The EULA of the entity must be accepted for it to be instantiated (request id: 6526c7f4-ab57-451a-a53b-10ebc0b22d7a)
```
Looking at the `cse.log`(URL was changed for the post):

```
11-16 19:20:15 urllib3.connectionpool 393 DEBUG    https://labvcd.example.net:443 "POST /api/vApp/vapp-98c97af1-9692-4471-8be7-e23dd61ea315/action/recomposeVApp HTTP/1.1" 400 None
11-16 19:20:15 cse.broker   490 ERROR    Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/container_service_extension/broker.py", line 449, in create_cluster_thread
    self.client_tenant, org, vdc, vapp, self.body)
  File "/usr/lib/python3.6/site-packages/container_service_extension/cluster.py", line 156, in add_nodes
    task = vapp.add_vms(specs, power_on=not reconfigure_hw)
  File "/usr/lib/python3.6/site-packages/pyvcloud/vcd/vapp.py", line 865, in add_vms
    EntityType.RECOMPOSE_VAPP_PARAMS.value, params)
  File "/usr/lib/python3.6/site-packages/pyvcloud/vcd/client.py", line 1088, in post_linked_resource
    media_type)
  File "/usr/lib/python3.6/site-packages/pyvcloud/vcd/client.py", line 1072, in post_resource
    objectify_results=objectify_results)
  File "/usr/lib/python3.6/site-packages/pyvcloud/vcd/client.py", line 861, in _do_request
    response), _objectify_response(response, objectify_results))
  File "/usr/lib/python3.6/site-packages/pyvcloud/vcd/client.py", line 866, in _response_code_to_exception
    raise BadRequestException(sc, request_id, objectify_response)
pyvcloud.vcd.exceptions.BadRequestException: Status code: 400/EULA_NOT_ACCEPTED, The EULA of the entity must be accepted for it to be instantiated (request id: 6526c7f4-ab57-451a-a53b-10ebc0b22d7a)
```

Im also seeing this message after just about every GET/POST in the `cse.log`:

```
11-16 19:20:12 urllib3.connectionpool 240 DEBUG    Resetting dropped connection: labvcd.example.net
```
I dont think that would have anything to do with the error im seeing but seemed note worthy at least.

**Comment by frenchtoasters** [comment]
@rocknes here you go

```
root@photon-machine [ ~/cse ]# pip list
Package                     Version   
--------------------------- ----------
asn1crypto                  0.24.0    
cachetools                  3.0.0     
certifi                     2018.10.15
cffi                        1.11.5    
chardet                     3.0.4     
Click                       7.0       
cloud-init                  0.7.9     
colorama                    0.4.0     
configobj                   5.0.6     
container-service-extension 1.2.2     
cryptography                2.4.1     
entrypoints                 0.2.3     
flufl.enum                  4.1.1     
humanfriendly               4.17      
idna                        2.7       
jeepney                     0.4       
Jinja2                      2.9.5     
jsonpatch                   1.15      
jsonpointer                 1.10      
keyring                     12.0.0    
lxml                        4.2.5     
MarkupSafe                  1.0       
motdgen                     0.1.3     
oauthlib                    2.0.2     
pika                        0.12.0    
pip                         18.1      
prettytable                 0.7.2     
pycparser                   2.19      
pycryptodome                3.4.11    
Pygments                    2.2.0     
pyvcloud                    20.0.1    
pyvmomi                     6.7.1     
PyYAML                      3.12      
requests                    2.20.1    
SecretStorage               3.1.0     
setuptools                  39.0.1    
six                         1.10.0    
tabulate                    0.8.2     
urllib3                     1.24.1    
vcd-cli                     21.0.0    
vsphere-guest-run           0.0.7     
```

Also here is the OS release information:

```
root@photon-machine [ ~/cse ]# cat /etc/photon-release 
VMware Photon OS 2.0
PHOTON_BUILD_NUMBER=304b817
```

**Comment by rocknes** [comment]
Thanks, seems like you are already on the latest version of the peojects. I just looked at the code where the failure is originating from and it looks like CSE is not setting the flag accept_all_eulas while invoking vapp.add_vm(). I will update here once I find more.

**Comment by frenchtoasters** [comment]
@rocknes so this looks like only an issue with the default template I had set as I was able to select the ubuntu template and have the cluster get passed this state and actually deploy the cluster. 

```
root@photon-machine [ ~ ]# vcd cse template list
InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised.
catalog    catalog_item                            description        is_default    name
---------  --------------------------------------  -----------------  ------------  ------------
cse        photon-custom-hw11-2.0-304b817-k8s      PhotonOS v2        True          photon-v2
                                                   Docker 17.06.0-4
                                                   Kubernetes 1.9.1
                                                   weave 2.3.0
cse        ubuntu-16.04-server-cloudimg-amd64-k8s  Ubuntu 16.04       False         ubuntu-16.04
                                                   Docker 18.03.0~ce
                                                   Kubernetes 1.10.1
                                                   weave 2.3.0
```

**Comment by mgaruccio** [comment]
I've had this happen in cases where the template customization process (which accepts the eula) didn't properly complete but the installation captured a template anyway.  You may want to try re-running the install and monitoring the output for errors or warnings.

**Comment by andrew-ni** [comment]
Hi @frenchtoasters , as @mgaruccio said, this EULA problem should only exist if template creation/customization failed at some point, but was still captured as a vapp template.

Could you try re-installing CSE to recreate these templates?

Note: In 1.2.2 some users are experiencing misplaced CSE script files (Data file not found! error) which can cause to failed template customization. For workaround, check out https://github.com/vmware/container-service-extension/issues/145#issuecomment-434964458 (will be fixed in 1.2.3, which should be pushed to PyPI this week.
